Entities/Common/Controls/StandardControlsCommon.as

```diff
@@ -40,3 +40,61 @@ bool isTapPickup(CBlob@ this, int ticks = 15)
 	// TODO: merge some code with the above and make it generalized to all keys if ever useful
 	return (getGameTime() - this.get_s32("tap_pickup_time") < ticks);
 }
+
+
+bool PointInsidePolygon( Vec2f Point, Vec2f[] polyVerts, Vec2f polyPos, bool facingLeft )
+{
+	// Mirror the polygon when the blob is facing left
+	// We make a copy, as we do not want to modify the original
+    Vec2f[] polygon;
+
+    if (facingLeft) 
+    {
+      Vec2f[] mirrored;
+      for ( int i = 0 ; i < polyVerts.length ; i++ )
+      {
+        Vec2f q = Vec2f(-polyVerts[i].x, polyVerts[i].y);
+        mirrored.push_back(q);
+      }
+
+      polygon = mirrored;
+    } 
+    else 
+    {
+      polygon = polyVerts;
+    }
```

Comment:
Scrap the local variable. `Vec2f[] polyVerts` is already passed by value, so the original will be untouched anyway.

```suggestion
bool PointInsidePolygon( Vec2f Point, Vec2f[] polygon, Vec2f polyPos, bool facingLeft )
{
    // Mirror the polygon when the blob is facing left
    if (facingLeft) 
    {
      for ( int i = 0 ; i < polygon.length ; i++ )
      {
        polygon[i].x = -polygon[i].x;
      }
    }
```